Jenkinsfile

```diff
@@ -8,9 +8,23 @@ elifePipeline {
     node('containers-jenkins-plugin') {
         stage 'Build images', {
             checkout scm
+            sh "find build/critical-css -name '*.css' -type f -delete"
+            sh "touch build/critical-css/default.css"
             dockerComposeBuild commit
         }
 
+        stage 'Generate critical CSS', {
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml up -d critical_css"
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml exec -T critical_css node_modules/.bin/gulp critical-css:generate"
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml exec -T critical_css ./check_critical_css.sh"
+            sh "docker cp journal_critical_css_1:build/critical-css/. build/critical-css/"
+            sh "docker-compose -f docker-compose.yml -f docker-compose.ci.yml down -v"
+        }
+
+        stage 'Rebuild FPM image', {
+            sh "IMAGE_TAG=${commit} docker-compose -f docker-compose.yml -f docker-compose.ci.yml build fpm"
+        }
+
```

Comment:
the ordering could be considered a bit strange as it separates the build and test phases, but doesn't allow feedback of unit tests until the full set of critical CSS is built. It makes sense to execute tests on the final image that will be published so in principle is correct but adds 1'30'' to the feedback loop. If the critical CSS are truly an optimization and not a requirement, this could be moved after `Project tests`?